File: backend/app/schemas/predict.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

# Overload Prediction
class OverloadPredictionRequest(BaseModel):
    facility_ids: List[str]
    time_window_days: int = 7
    include_reasons: bool = True

class OverloadRisk(BaseModel):
    facility_id: str
    risk_level: str  # "Safe", "Watch", "Critical"
    predicted_overload_days: int
    reasons: List[str]

class OverloadPredictionResponse(BaseModel):
    predictions: List[OverloadRisk]
    timestamp: datetime

# Outbreak Severity
class OutbreakSeverityRequest(BaseModel):
    area: str
    disease: str
    recent_reports: int
    lab_confirmations: int = 0
    population_density: float = 0.0

class OutbreakSeverityResponse(BaseModel):
    severity: str  # "Low", "Medium", "High"
    confidence: str  # "Low", "Medium", "High"
    reasons: List[str]
    timestamp: datetime

# KPI Snapshot
class KPISnapshot(BaseModel):
    facilities_at_risk: int
    avg_bed_occupancy: float
    icu_occupancy: float
    opd_today: int
    avg_travel_time: float
    active_outbreaks: int
    mobile_units_active: int
    timestamp: datetime

File: backend/rust-predictions-api/src/schemas/predict.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
pub struct OverloadPredictionRequest {
    pub facility_ids: Vec<String>,
    pub time_window_days: Option<i32>,
    pub include_reasons: Option<bool>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct OverloadRisk {
    pub facility_id: String,
    pub risk_level: String,
    pub predicted_overload_days: i32,
    pub reasons: Vec<String>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct OverloadPredictionResponse {
    pub predictions: Vec<OverloadRisk>,
    pub timestamp: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct OutbreakSeverityRequest {
    pub area: String,
    pub disease: String,
    pub recent_reports: i32,
    pub lab_confirmations: Option<i32>,
    pub population_density: Option<f64>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct OutbreakSeverityResponse {
    pub severity: String,
    pub confidence: String,
    pub reasons: Vec<String>,
    pub timestamp: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct KPISnapshot {
    pub facilities_at_risk: i32,
    pub avg_bed_occupancy: f64,
    pub icu_occupancy: f64,
    pub opd_today: i32,
    pub avg_travel_time: f64,
    pub active_outbreaks: i32,
    pub mobile_units_active: i32,
    pub timestamp: String,
}